client/app/bundles/course/assessment/pages/AssessmentMonitoring/components/SessionDetailsPopup.tsx

```diff
@@ -64,6 +68,16 @@ const SessionDetailsPopup = (props: SessionDetailsPopupProps): JSX.Element => {
           </Typography>
         </header>
 
+        {props.submissionId && (
+          <Link
+            className="px-2 !mt-0 !mb-4"
+            opensInNewTab
+            to={`/courses/${courseId}/assessments/${assessmentId}/submissions/${props.submissionId}/edit`}
```

Comment:
Should we start to create routes builder in our FE?

Reply:
It's actually not trivial to have this.

React Router provides a built-in way of generating paths "safely" with [`generatePath`](https://reactrouter.com/en/main/utils/generate-path). But this only infers the route parameters. It doesn't provide any safety with regards to our own routers, and we still have to type out the raw path string.

![image](https://github.com/Coursemology/coursemology2/assets/51525686/f914b070-5f94-4331-827d-c8fc2a104cc8)

I found a library that aims for type-safe routes in React Router: [react-router-typesafe-routes](https://github.com/fenok/react-router-typesafe-routes), but integrating it may require changes in many components, especially if we are replacing `useParams` with their `useTypedParams`.